src/main/java/uk/co/ramp/AppConfig.java

```diff
@@ -80,35 +105,53 @@ public PopulationProperties populationProperties() {
   }
 
   @Bean
-  public RandomDataGenerator randomDataGenerator(
-      @Value("${cmdLineArgument:#{null}}") Optional<String[]> argumentValue) {
-    long arg = 0;
+  public AgeRetriever ageRetriever(
+      PopulationProperties populationProperties, RandomDataGenerator randomDataGenerator) {
+    try (Reader reader = getReader(inputFiles().ageData())) {
+      var ageData = new AgeDataReader().read(reader);
+      return new AgeRetriever(populationProperties, randomDataGenerator, ageData);
+    } catch (IOException e) {
+      String message = "An error occurred while parsing the age data at " + inputFiles().ageData();
+      LOGGER.error(message);
+      throw new ConfigurationException(message, e);
+    }
+  }
+
+  private void useSeed(RandomDataGenerator rdg, int seed) {
+    rdg.reSeed(seed);
+    LOGGER.info("Additional Seed information provided, the seed will be {}", seed);
+  }
+
+  private void logUsingDefaultSeed() {
+    LOGGER.info("Additional Seed information not provided, using internal random seed.");
+  }
+
+  @Bean
+  public RandomDataGenerator randomDataGenerator() {
+    RandomDataGenerator rdg = new RandomDataGenerator();
+
     try {
-      if (argumentValue.isPresent() && argumentValue.get().length > 0) {
-        arg = Integer.parseInt(argumentValue.get()[0]);
-        LOGGER.info(
-            "Additional Seed information provided, the seed will be {}",
-            standardProperties().seed() + arg);
-      } else {
-        LOGGER.info(
-            "Additional Seed information not provided, defaulting to {}",
-            standardProperties().seed());
-      }
-      RandomDataGenerator r = new RandomDataGenerator();
-      r.reSeed(standardProperties().seed() + arg);
-      return r;
+      OptionalInt cmdLineSeedOverride =
+          Stream.ofNullable(seed).mapToInt(Integer::parseInt).findAny();
+      OptionalInt propertiesSeedOverride = standardProperties().seed();
+      Stream.of(cmdLineSeedOverride, propertiesSeedOverride)
+          .flatMapToInt(OptionalInt::stream)
+          .reduce(Integer::sum)
+          .ifPresentOrElse(seed -> useSeed(rdg, seed), this::logUsingDefaultSeed);
```

Comment:
This is quite different to the style I'm used to (but I'm not complaining)!
I would have done something like:
```
if(seed == null && !standardProperties().seed().isPresent()) {
    logUsingDefaultSeed();
} else {
    int cmdLineSeed = seed == null ? 0 : Integer.parseInt(seed);
    int propertiesSeed = standardProperties().seed().orElse(0);
    useSeed(rdg, seed);
}
```

Reply:
Seems like I've overused the notion of streams here, definitely simpler to read your version. Will update. Thank you!

Reply:
Thanks!